lib/automatic/api_client/client_delegator.rb

```diff
@@ -0,0 +1,21 @@
+module Automatic
+  class ApiClient
+    class ClientDelegator
+      attr_reader :api_client, :delegated_class
+
+      def initialize(api_client, delegated_class)
+        @api_client = api_client
+        @delegated_class = delegated_class
+      end
+
+      def method_missing(method_sym, *arguments, &block)
+        super unless delegated_class.respond_to?(method_sym)
+
+        arguments[1 + delegated_class.method(method_sym).arity] ||= {}
```

Comment:
I wasn't sure how else to insert a key-value into the options hash when calling the method in the delegated_class. `arguments.last` doesn't work since the options hash is optional, and the methods also have varying numbers of arguments.

The alternative is to make `trips.rb` and `vehicles.rb` more verbose and know which individual methods in `Automatic::Models::Trips` need to have the instance-level client passed in. Do-able, but that ties these classes together closer than I'd like. Very open to other potential solutions if you've got any